msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ForumScheduleNotificationImpl.java

```diff
@@ -28,245 +29,241 @@
-    			makeAvailable = true;
-    		}
-    		
-    		
-			if(openTime != null && openTime.before(timeService.newTime())){
+			Instant openTime = null;
+			Instant closeTime = null;
+			if (openDate != null) {
+				openTime = openDate.toInstant();
+			}
+			if (closeDate != null) {
+				closeTime = closeDate.toInstant();
+			}
+			if (closeDate == null && openDate == null) {
+				// user didn't specify either, so open topic
+				makeAvailable = true;
+			}
+
+			if (openTime != null && openTime.isBefore(Instant.now())) {
 				afterOpen = true;
-			}else if(openTime == null){
+			} else if (openTime == null) {
 				afterOpen = true;
 			}
-			if(closeTime != null && closeTime.after(timeService.newTime())){
+			if (closeTime != null && closeTime.isAfter(Instant.now())) {
 				beforeClose = true;
-			}else if(closeTime == null){
+			} else if (closeTime == null) {
 				beforeClose = true;
 			}
 
-			if(afterOpen && beforeClose){
-				makeAvailable = true;    				
+			if (afterOpen && beforeClose) {
+				makeAvailable = true;
 			}
 		}
 		return makeAvailable;
-    }
-    
-    public void updateSynopticMessagesForForumComparingOldMessagesCount(String siteId, Long forumId, Long topicId, HashMap<String, Integer> beforeChangeHM, int numOfAttempts) {
-  	  try {
-  		  // update synotpic info for forums only:
-  		  SynopticMsgcntrManagerCover
-  		  .updateSynopticMessagesForForumComparingOldMessagesCount(
-  				  siteId, forumId, topicId, beforeChangeHM);
-  	  } catch (HibernateOptimisticLockingFailureException holfe) {
+	}
 
-  		  // failed, so wait and try again
-  		  try {
-  			  Thread.sleep(SynopticMsgcntrManager.OPT_LOCK_WAIT);
-  		  } catch (InterruptedException e) {
-  			  e.printStackTrace();
-  		  }
+	public void updateSynopticMessagesForForumComparingOldMessagesCount(String siteId, Long forumId, Long topicId,
+			HashMap<String, Integer> beforeChangeHM, int numOfAttempts) {
+		try {
+			// update synotpic info for forums only:
+			SynopticMsgcntrManagerCover.updateSynopticMessagesForForumComparingOldMessagesCount(siteId, forumId,
+					topicId, beforeChangeHM);
+		} catch (HibernateOptimisticLockingFailureException holfe) {
 
-  		  numOfAttempts--;
+			// failed, so wait and try again
+			try {
+				Thread.sleep(SynopticMsgcntrManager.OPT_LOCK_WAIT);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+
+			numOfAttempts--;
 
-  		  if (numOfAttempts <= 0) {
-  			  System.out
-  			  .println("ForumScheduleNotificationImpl: HibernateOptimisticLockingFailureException no more retries left");
-  			  holfe.printStackTrace();
-  		  } else {
-  			  System.out
-  			  .println("ForumScheduleNotificationImpl: HibernateOptimisticLockingFailureException: attempts left: "
-  					  + numOfAttempts);
-  			  updateSynopticMessagesForForumComparingOldMessagesCount(siteId,
-  					  forumId, topicId, beforeChangeHM, numOfAttempts);
-  		  }
-  	  }
-    }
+			if (numOfAttempts <= 0) {
+				System.out.println(
```

Comment:
should be a log message, lets continue to remove the System.out.print

Reply:
OK